pkg/api/mlflow/service/artifact/storage/s3.go

```diff
@@ -69,33 +69,32 @@ func (s S3) List(artifactURI, path, nextPageToken string) (string, string, []Art
 	// 1. process search `prefix` parameter.
 	path, err = url.JoinPath(*input.Prefix, path)
 	if err != nil {
-		return "", "", nil, eris.Wrap(err, "error constructing s3 prefix")
+		return "", nil, eris.Wrap(err, "error constructing s3 prefix")
 	}
 	input.Prefix = aws.String(path)
 
-	// 2. process search `nextPageToken` parameter.
-	if nextPageToken != "" {
-		input.ContinuationToken = aws.String(nextPageToken)
-	}
-
-	output, err := s.client.ListObjectsV2(context.TODO(), &input)
-	if err != nil {
-		return "", "", nil, eris.Wrap(err, "error getting s3 objects")
-	}
-
-	log.Debugf("got %d objects from S3 storage for path: %s", len(output.Contents), path)
-	artifactList := make([]ArtifactObject, len(output.Contents))
-	for i, object := range output.Contents {
-		artifactList[i] = ArtifactObject{
-			Path:  *object.Key,
-			Size:  object.Size,
-			IsDir: false,
+	var artifactList []ArtifactObject
+	for {
+		output, err := s.client.ListObjectsV2(context.TODO(), &input)
+		if err != nil {
+			return "", nil, eris.Wrap(err, "error getting s3 objects")
+		}
+		log.Debugf("got %d objects from S3 storage for path: %s", len(output.Contents), path)
+		for _, object := range output.Contents {
+			artifactList = append(artifactList, ArtifactObject{
+				Path:  *object.Key,
+				Size:  object.Size,
+				IsDir: false,
+			})
 		}
-	}
 
-	if output.NextContinuationToken != nil {
-		return *output.NextContinuationToken, s.config.ArtifactRoot, artifactList, nil
+		// if NextContinuationToken exists, then iterate over other pages until we process all the results.
+		if output.NextContinuationToken != nil {
+			input.ContinuationToken = output.NextContinuationToken
+		} else {
+			break
+		}
```

Comment:
please use `s3.NewListObjectsV2Paginator` instead, and then loop with
```golang
for paginator.HasMorePages() {
	output, err := paginator.NextPage(context.TODO())
	...
}
```

Reply:
got it. done.